docs/streams/core-concepts.html

```diff
@@ -172,6 +172,35 @@ <h2><a id="streams_processing_guarantee" href="#streams_processing_guarantee">Pr
         More details can be found in the <a href="/{{version}}/documentation#streamsconfigs"><b>Kafka Streams Configs</b></a> section.
     </p>
 
+    <h3><a id="streams_out_of_ordering" href="#streams_out_of_ordering">Out-of-Ordering Handling</a></h3>
+
+    <p>
+        Besides the guarantee that each record will be processed exactly-once, another issue that many stream processing application will face is how to
+        handle <a href="tbd">out-of-ordering data</a> data that may impact their business logic. In Kafka Streams, there are two causes that could potentially
+        result in out-of-ordering data arrivals with respect to their timestamps:
+    </p>
+
+    <ul>
+        <li> Within a topic-partition, record's timestamp may not be monotonically increasing along with their offsets. Since Kafka Streams will always try to process records within a topic-partition to follow the offset ordering, it can cause records with larger timestamps
+             to be processed earlier than records with smaller timestamp in the same topic-partition.</li>
+        <li> Within a <a href="/{{version}}/documentation/streams/architecture#streams_architecture_tasks">stream task</a> that may be processing multiple topic-partitions, if users configure the application to not wait for all partitions contain some buffered data and
```

Comment:
This part skips over the fact that users have a choice in configuring the application to wait for all partitions to contain data or not.  It might be worth a sentence or two explaining that configuration.

Reply:
I've talked about this in the latter paragraph.